### PR TITLE
[Backend][UI] As a User, I can sign out of my account

### PR DESCRIPTION
**Pull request by Goose97**
Closes [[Backend] As a User, I can sign out of my account](https://github.com/Goose97/google-scraper-ruby/issues/53)
Closes [[UI] As a User, I can sign out of my account](https://github.com/Goose97/google-scraper-ruby/issues/54)

## What happened 👀

Implement sign out feature

## Insight 📝

- [x] Display sign out button in every page (except sign in/sign up pages)
- [x] After sign out successfully, redirect users to sign in page

## Proof Of Work 📹

https://github.com/Goose97/google-scraper-ruby/assets/35915460/5659a670-36bb-4a08-8130-8e3a77982fcc

### CODE REVIEW

**Comment by olivierobert** [comment]
You need to check why and fix the tests are failing:

image

File: app/assets/stylesheets/layouts/_application.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.app-header {
```
Comment (olivierobert) [inline app/assets/stylesheets/layouts/_application.scss]: All layouts must be namespaced to allow other layouts to be used:

```
.layout-application {
  .app-header {

  }
}
```

That class must be added to the top most HTML element which is the HTML tag e.g., https://github.com/nimblehq/mia-web/blob/development/engines/mia_campaign/app/views/layouts/mia_campaign/application.html.slim#L2

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<h2><%= I18n.t('auth.sign_in') %></h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
```
Comment (olivierobert) [inline app/views/devise/sessions/new.html.erb]: Same issue before with using `form_with`

**Comment by olivierobert** [comment]
Nevermind for the tests, I saw https://github.com/Goose97/google-scraper-ruby/pull/52#issuecomment-1713504841

File: spec/systems/users/view_sign_in_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'View sign in page', type: :system do
+  context 'given an unathenticated user' do
```
Comment (olivierobert) [inline spec/systems/users/view_sign_in_page_spec.rb]: Typo:

```suggestion
  context 'given an unauthenticated user' do
```

File: app/assets/stylesheets/layouts/_application.scss (hunk 1 above)
Comment (Goose97) [reply app/assets/stylesheets/layouts/_application.scss]: Thanks for the tip. Fixed in e397a6be9e3661d6e0630cef612168987bcbe423

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (Goose97) [reply app/views/devise/sessions/new.html.erb]: Fixed in 559bb4a367e391ec4cdfe8eb6c8f929167def53e

File: spec/systems/users/view_sign_in_page_spec.rb (hunk 3 above)
Comment (Goose97) [reply spec/systems/users/view_sign_in_page_spec.rb]: Fixed in ac6eaf2feef7c5ac21a171dd692a4e7d84bc32b3

**Review by olivierobert — APPROVED** [review]

File: app/assets/stylesheets/layouts/_application.scss
Comment (olivierobert) [inline app/assets/stylesheets/layouts/_application.scss:1]: The file name should then be: `_default.scss` to sync the class name and the file structure `layouts/_default.scss` for easier discoverability.

**Review by andyduong1920 — APPROVED** [review]
apart from Olivier's comment about the `layout` naming, the rest LGTM 👍

**Pull request merged**